packages/patterns/src/components/partial-syncing-controls.js

```diff
@@ -1,108 +1,106 @@
-/**
- * External dependencies
- */
-import { nanoid } from 'nanoid';
-
 /**
  * WordPress dependencies
  */
-import { InspectorControls } from '@wordpress/block-editor';
-import { BaseControl, CheckboxControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
+import { usePrevious } from '@wordpress/compose';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { PARTIAL_SYNCING_SUPPORTED_BLOCKS } from '../constants';
 
-function PartialSyncingControls( { name, attributes, setAttributes } ) {
-	const syncedAttributes = PARTIAL_SYNCING_SUPPORTED_BLOCKS[ name ];
-	const attributeSources = syncedAttributes.map(
-		( attributeName ) =>
-			attributes.metadata?.bindings?.[ attributeName ]?.source
-	);
-	const isConnectedToOtherSources = attributeSources.every(
-		( source ) => source && source !== 'core/pattern-overrides'
-	);
+function removeBindings( bindings, syncedAttributes ) {
+	let updatedBindings = {};
+	for ( const attributeName of Object.keys( syncedAttributes ) ) {
+		// Omit any pattern override bindings from the `updatedBindings` object.
+		if (
+			bindings?.[ attributeName ]?.source !== 'core/pattern-overrides' &&
+			bindings?.[ attributeName ]?.source !== undefined
+		) {
+			updatedBindings[ attributeName ] = bindings[ attributeName ];
+		}
+	}
+	if ( ! Object.keys( updatedBindings ).length ) {
+		updatedBindings = undefined;
+	}
+	return updatedBindings;
+}
 
-	// Render nothing if all supported attributes are connected to other sources.
-	if ( isConnectedToOtherSources ) {
-		return null;
+function setBindings( bindings, syncedAttributes ) {
+	const updatedBindings = { ...bindings };
+	for ( const attributeName of Object.keys( syncedAttributes ) ) {
+		if ( ! bindings?.[ attributeName ] ) {
+			updatedBindings[ attributeName ] = {
+				source: 'core/pattern-overrides',
+			};
+		}
 	}
+	return updatedBindings;
+}
+
+export default function useSetPatternBindings(
+	{ name, attributes, setAttributes },
+	currentPostType
+) {
+	const metadataName = attributes?.metadata?.name;
+	const prevMetadataName = usePrevious( metadataName ) ?? metadataName;
+	const bindings = attributes?.metadata?.bindings;
+
+	useEffect( () => {
+		// Bindings should only be created when editing a wp_block post type,
+		// and also when there's a change to the user given name for the block.
```

Comment:
```suggestion
		// and also when there's a change to the user-given name for the block.
```

Tiniest of tiny nits 🙂